src/serializer/ResidualHeapVisitor.js

```diff
@@ -648,11 +666,9 @@ export class ResidualHeapVisitor {
       );
       return getOrDefault(residualFunctionBindings, name, (): ResidualFunctionBinding => {
         invariant(environment instanceof DeclarativeEnvironmentRecord);
-        let binding = environment.bindings[name];
-        invariant(binding !== undefined);
-        invariant(!binding.deletable);
         return {
-          value: (binding.initialized && binding.value) || this.realm.intrinsics.undefined,
+          name,
+          value: undefined,
```

Comment:
Please add a comment why in the global case we visit immediately but in this case, we deliay initializing the residual function binding.

Reply:
Adding comment.